src/live.rs

```diff
@@ -158,6 +162,163 @@ pub fn pxe_ignition_unwrap(config: &PxeIgnitionUnwrapConfig) -> Result<()> {
+    write_iso_kargs(&mut holder, area_size, &area_offsets, &default_kargs)
+}
+
+pub fn iso_kargs_show(config: &IsoKargsShowConfig) -> Result<()> {
+    let mut holder = CopiedFileHolder::new(&config.input, None)?;
+    let (area_size, default_kargs_offset, area_offsets) = find_karg_embed_areas(&mut holder.file)?;
+    let kargs = if config.default {
+        get_kargs_at_offset(&mut holder, area_size, default_kargs_offset)?
+    } else {
+        get_current_kargs(&mut holder, area_size, &area_offsets)?
+    };
+    println!("{}", kargs);
+    Ok(())
+}
+
+fn get_current_kargs(
+    holder: &mut CopiedFileHolder,
+    area_size: usize,
+    area_offsets: &[u64],
+) -> Result<String> {
+    // really, we could just get the kargs from the first file, but let's sanity-check that all the
+    // offsets have the same kargs
+    let mut first_kargs: Option<String> = None;
+    for offset in area_offsets {
+        let kargs = get_kargs_at_offset(holder, area_size, *offset)?;
+        if let Some(ref first_kargs) = first_kargs {
+            if &kargs != first_kargs {
+                bail!("kargs don't match at all offsets!");
+            }
+        } else {
+            first_kargs = Some(kargs);
+        }
+    }
+    Ok(first_kargs.expect("at least one area offset"))
+}
+
+fn get_kargs_at_offset(
+    holder: &mut CopiedFileHolder,
+    area_size: usize,
+    offset: u64,
+) -> Result<String> {
+    holder
+        .file
+        .seek(SeekFrom::Start(offset))
+        .chain_err(|| "seeking to karg area offset")?;
+    let area = {
+        let mut buf = vec![0u8; area_size];
+        holder
+            .file
+            .read_exact(&mut buf)
+            .chain_err(|| "reading karg area")?;
+        String::from_utf8(buf).chain_err(|| "invalid UTF-8 in karg area")?
+    };
+    Ok(area.trim_end_matches('#').trim().into())
+}
+
+fn write_iso_kargs(
+    holder: &mut CopiedFileHolder,
+    area_size: usize,
+    area_offsets: &[u64],
+    kargs: &str,
+) -> Result<()> {
+    let kargs: String = kargs.trim().to_string() + "\n";
+    if kargs.len() > area_size {
+        bail!("kargs too large for area: {} vs {}", kargs.len(), area_size);
+    }
+
+    let mut new_area = vec![b'#'; area_size];
+    new_area[..kargs.len()].copy_from_slice(kargs.as_bytes());
+
+    for &area_offset in area_offsets {
+        holder
+            .file
+            .seek(SeekFrom::Start(area_offset))
+            .chain_err(|| "seeking to karg area offset")?;
+        holder
+            .file
+            .write_all(&new_area)
+            .chain_err(|| "writing karg embed area")?;
+    }
+
+    holder.complete();
+    Ok(())
+}
+
+fn find_karg_embed_areas(file: &mut File) -> Result<(usize, u64, Vec<u64>)> {
+    let mut buf: [u8; 8] = [0; 8];
+    // The ISO 9660 System Area is 32 KiB. Karg embed area information is located in the 72 bytes
+    // before the initrd embed area (see EmbedArea below):
+    // 8 bytes: magic string "corekarg"
```

Comment:
https://github.com/coreos/coreos-assembler/pull/1996 changes this to `coreKarg`.